analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Binary.cs

```diff
@@ -113,6 +117,7 @@ kind switch
             {
                 BinaryOperatorKind.Equals or BinaryOperatorKind.ObjectValueEquals => BoolConstraint.From(isNullLeft && isNullRight),
                 BinaryOperatorKind.NotEquals or BinaryOperatorKind.ObjectValueNotEquals => BoolConstraint.From(isNullLeft != isNullRight),
+                BinaryOperatorKind.GreaterThan or BinaryOperatorKind.GreaterThanOrEqual or BinaryOperatorKind.LessThan or BinaryOperatorKind.LessThanOrEqual => BoolConstraint.False,
```

Comment:
I find this easier to read and it follows DRY. WDYT?
```suggestion
                { } when kind.IsAnyRelational() => BoolConstraint.False,
```

Reply:
I used `_ when` instead of `{ } when`, since `kind` is an `enum` and, as such, always non-null.
And if a variable cannot be null, pattern matching should not check for `{ }`.